File: world-cup-system/src/main/java/com/project/worldcupsystem/models/Stadium.java
package com.project.worldcupsystem.models;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "stadium")
public class Stadium {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long stadiumId;
    private String name;
    private String shape;
    private Long seatsCount;
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "vip_lounge_id", referencedColumnName = "id")
    private VipLounge vipLounge;
    @OneToMany
    private List<Match> matches;


    public Stadium() {
    }

    public Stadium(String name, String shape, Long seatsCount) {
        this.name = name;
        this.shape = shape;
        this.seatsCount = seatsCount;
    }

    public Long getStadiumId() {
        return stadiumId;
    }

    public void setStadiumId(Long stadiumId) {
        this.stadiumId = stadiumId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getShape() {
        return shape;
    }

    public void setShape(String shape) {
        this.shape = shape;
    }

    public Long getSeatsCount() {
        return seatsCount;
    }

    public void setSeatsCount(Long seatsCount) {
        this.seatsCount = seatsCount;
    }

    public List<Match> getStadiumMatches() {
        return matches;
    }

    public void setStadiumMatches(List<Match> stadiumMatches) {
        this.matches = stadiumMatches;
    }
}

File: world_cup_system_frontend_web/lib/data_models/stadium.dart
// class Stadium {
//   final id;
//   final name;
//   final shape;
//   final seatsCount;
//   final vipLounge;
//   final stadiumslist;

//   Stadium({
//     this.id,
//     this.name,
//     this.shape,
//     this.seatsCount,
//     this.vipLounge,
//     this.stadiumslist,
//   });

//   static fromJson(responseItem) {
//     return Stadium(
//       id: responseItem['id'],
//       name: responseItem['name'],
//       shape: responseItem['shape'],
//       seatsCount: responseItem['seatsCount'],
//       vipLounge: responseItem['vipLounge'],
//     );
//   }

//   // convert to json
//   Map<String, dynamic> toJson() {
//     return {
//       'id': id,
//       'name': name,
//       'shape': shape,
//       'seatsCount': seatsCount,
//       'vipLounge': vipLounge,
//     };
//   }
// }
